docs/kubernetes-reliability.md

```diff
@@ -0,0 +1,101 @@
+Distributed system such as Kubernetes are designed to be resilient to the
+failures.  More details about Kubernetes High-Availability (HA) may be found at
+[Building High-Availability Clusters](https://kubernetes.io/docs/admin/high-availability/)
+
+To have a simple view the most of parts of HA will be skipped to describe
+Kubelet<->Controller Manager communication only.
+
+By default the normal behavior looks like:
+
+1. Kubelet updates its status to Master's API Server periodically described at
+   `-–node-status-update-frequency`.
+
+2. Kubernetes controller manager checks the statuses of Kubelets every
+   `–-node-monitor-period`.
+
+3. In case the status is updated  within `--node-monitor-grace-period` of time,
+   Kubernetes controller manager considers healthy status of Kubelet.
+
+> Kubernetes controller manager and Kubelets work asynchronously. It means that
+> the delay may include network latency, API Server latency, etcd latency,
+> latency caused by load on one's master nodes and so on. So if
+> `--node-status-update-frequency` is set to 5s in reality it may appear in
+> etcd in 6-7 seconds or even longer when etcd cannot commit data to quorum
+> nodes.
+
+# Failure
+
+Kubelet will try to make `nodeStatusUpdateRetry` post attempts. Currently
```

Comment:
This is a bit above what an average user needs to know, but ok. I would just say Kubelet tries 5 times to contact apiserver. It's not configurable.